addons/website/models/res_users.py

```diff
@@ -72,18 +72,18 @@ def authenticate(cls, db, login, password, user_agent_env):
             visitor_pre_authenticate_sudo = request.env['website.visitor']._get_visitor_from_request()
         uid = super(ResUsers, cls).authenticate(db, login, password, user_agent_env)
         if uid and visitor_pre_authenticate_sudo:
-            with cls.pool.cursor() as cr:
-                env = api.Environment(cr, uid, {})
-                user_partner = env.user.partner_id
-                visitor_current_user_sudo = env['website.visitor'].sudo().search([
-                    ('partner_id', '=', user_partner.id)
-                ], limit=1)
-                if visitor_current_user_sudo:
-                    # A visitor exists for the logged in user, link public
-                    # visitor records to it.
+            env_uid = api.Environment(visitor_pre_authenticate_sudo.env.cr, uid, {})
+            user_partner = env_uid.user.partner_id
+            visitor_current_user_sudo = env_uid['website.visitor'].sudo().search([
+                ('partner_id', '=', user_partner.id)
+            ], limit=1)
+            if visitor_current_user_sudo:
+                # A visitor exists for the logged in user, link public
+                # visitor records to it.
+                if visitor_pre_authenticate_sudo != visitor_current_user_sudo:
```

Comment:
I'd just keep this line in the PR if this is enough.

Reply:
Thank you for your help. I will only keep this line so and test again to ensure it solves the problem.

Reply:
Just to notify that the fix still works with the minimal change (just the added "if" statement to ensure the 2 visitors to be merged are different)

Reply:
Nice, just don't forget to reflect what I said in the commit message:
- This is not only about creating account but also login when already login
- This is due to going twice in the authenticate, when the second time we are already login

Kinda helpful ^^